spec.bs

```diff
@@ -1843,12 +1875,57 @@ Expand the section of the navigation/traversal response handling which deals wit
 
 <h3 id="navigate-event-traversal-patches">History traversal updates</h3>
 
+<div algorithm="check if unloading is user-canceled">
+  Modify the [=check if unloading is user-canceled=] algorithm as follows:
+
+  To [=check if unloading is user-canceled=] given a [=list=] of [=navigables=] |navigablesCrossingDocuments|, an optional integer |targetStep| (default null), and an optional [=user navigation involvement=] |userInvolvement| (default "<code>[=user navigation involvement/none=]</code>"):
+  1. Let |unloadPromptShown| be false.
+  1. Let |unloadPromptCanceled| be false.
+  1. Let |changingNavigables| be a [=list=] of [=navigables=], initially empty.
+  1. If |traversable| is not null and |targetStep| is not null, append the result of [=get all navigables whose current session history entry will change or reload=] given |traversable| and |targetStep| to |changingNavigables|.
+  1. Otherwise, append |navigablesCrossingDocuments| to |changingNavigables|.
+  1. Let |totalTasks| be the size of |changingNavigables|.
+  1. Let |completedTasks| be 0.
+  1. If |traversable| is not null and |changingNavigables| [=list/contains=] |traversable|:
+    1. Let |targetEntry| be the result of [=getting the target history entry=] given |traversable| and |targetStep|.
+    1. Let |isSameOrigin| be true if |targetEntry|'s [=session history entry/origin=] is the [=same origin|same=] as |traversable|'s [=navigable/current session history entry=]'s [=session history entry/origin=]; otherwise, false.
+    1. [=Queue a global task=] on the [=navigation and traversal task source=] given |traversable|'s [=navigable/active document=]'s [=relevant global object=] to run the following steps:
+      1. If |isSameOrigin| is true:
+        1. If the result of [=firing a traversal navigate event=] at |traversable|'s [=navigable/active window=]'s [=Window/navigation API=] with <i>[=fire a traversal navigate event/destinationEntry=]</i> set to |targetEntry| and <i>[=fire a traversal navigate event/userInvolvement=]</i> set to |userInvolvement| is false:
+        1. Set |unloadPromptCanceled| to true.
+      1. If |unloadPromptCanceled| is false and |navigablesCrossingDocuments| [=list/contains=] |traversable|:
+        1. Run the steps to fire a beforeunload event (currently steps 6.1 thru 6.8 in the [=check if unloading is user-canceled=] steps).
+      1. Increment |completedTasks|.
+    1. Wait for |completedTasks| to be 1.
+    1. If |unloadPromptCanceled| is true, then return "<code>refuse</code>";
+  1. For each |navigable| of |changingNavigables|:
+    1. If |navigable| equals |traversable|, then continue.
```

Comment:
```suggestion
    1. If |navigable| equals |traversable|, then [=continue=].
```

Reply:
Apparently we weren't referencing https://infra.spec.whatwg.org/ yet, so I added that. It's delightfully easy to link to new specs :)

Reply:
So it turns out it's just `[=iteration/continue=]`, and then it gets auto-added for you with no need to modify the anchors block :). Pushed that fixup.

Reply:
FYI the reason the anchors block is full of things right now is for basically two reasons:

- We are using a bunch of non-exported concepts from HTML. Using non-exported concepts requires adding them to the anchors block.
- We are using the rewrite, which is not in the linking database.

In most other cases, no anchors block additions should be required. (Although sometimes ReSpec specs need to be added, because ReSpec is problematic.)